src/krb5/_ccache.pyx

```diff
@@ -104,6 +120,38 @@ cdef extern from "python_krb5.h":
         krb5_ccache cache,
     ) nogil
 
+    int32_t KRB5_TC_MATCH_TIMES
+    int32_t KRB5_TC_MATCH_IS_SKEY
+    int32_t KRB5_TC_MATCH_FLAGS
+    int32_t KRB5_TC_MATCH_TIMES_EXACT
+    int32_t KRB5_TC_MATCH_FLAGS_EXACT
+    int32_t KRB5_TC_MATCH_AUTHDATA
+    int32_t KRB5_TC_MATCH_SRV_NAMEONLY
+    int32_t KRB5_TC_MATCH_2ND_TKT
+    int32_t KRB5_TC_MATCH_KTYPE
+    int32_t KRB5_TC_SUPPORTED_KTYPES
+
+
+_CredentialsRetrieveFlags_members = [
+    ('none', 0),
+    ('match_times', KRB5_TC_MATCH_TIMES),
+    ('match_is_skey', KRB5_TC_MATCH_IS_SKEY),
+    ('match_flags', KRB5_TC_MATCH_FLAGS),
+    ('match_times_exact', KRB5_TC_MATCH_TIMES_EXACT),
+    ('match_flags_exact', KRB5_TC_MATCH_FLAGS_EXACT),
+    ('match_authdata', KRB5_TC_MATCH_AUTHDATA),
+    ('match_srv_nameonly', KRB5_TC_MATCH_SRV_NAMEONLY),
+    ('match_2nd_tkt', KRB5_TC_MATCH_2ND_TKT),
+    ('match_keytype', KRB5_TC_MATCH_KTYPE),
+]
+# If KRB5_TC_SUPPORTED_KTYPES is not available it will be set to 0 in
+# python_krb5.h
+if KRB5_TC_SUPPORTED_KTYPES != 0:
+    _CredentialsRetrieveFlags_members += [
+        ('supported_ktypes', KRB5_TC_SUPPORTED_KTYPES),
+    ]
+CredentialsRetrieveFlags = enum.IntEnum('CredentialsRetrieveFlags', _CredentialsRetrieveFlags_members)
```

Comment:
Any reason why you are adding the members like this, the principal side just declared the enum like normal Python code https://github.com/jborean93/pykrb5/blob/2e5284a3076925fc8b8ab5d99ed89e8a0270986e/src/krb5/_principal.pyx#L77-L81.